kmodes/tests/test_kprototypes.py

```diff
@@ -333,3 +333,58 @@ def test_kprototypes_ninit(self):
         self.assertEqual(kmodes.n_init, 10)
         kmodes = kprototypes.KPrototypes(n_init=10, init=[np.array([]), np.array([])])
         self.assertEqual(kmodes.n_init, 1)
+
+    def test_kprototypes_sample_weights_validation(self):
+        kproto = kprototypes.KPrototypes(n_clusters=4, init='Cao', verbose=2)
+        sample_weights_too_few = [1] * 11
+        with self.assertRaises(ValueError):
```

Comment:
A small improvement would be to use `assertRaisesRegexp` and test the message itself too.

Reply:
True!
[03e9ac6](https://github.com/nicodv/kmodes/pull/171/commits/03e9ac68cbf923e8d53223def7e4f98fe542c802)